src/app/container/register-tool/register-tool.component.html

```diff
@@ -59,280 +59,274 @@ <h4 mat-dialog-title>Register Tool</h4>
-                placeholder="e.g. liftover-fast (optional)"
+                matTooltip="Custom Docker registry path"
+                placeholder="e.g. *-images.sbgenomics.com"
+                [pattern]="validationPatterns.sevenBridgesDockerRegistryPath"
               />
-              <div *ngIf="formErrors.toolName" class="alert alert-danger">{{ formErrors.toolName }}</div>
-            </div>
-          </div>
+              <mat-hint>Has the form *-images.sbgenomics.com</mat-hint>
+              <mat-error *ngIf="formErrors.sevenBridgesDockerRegistryPath">{{ formErrors.sevenBridgesDockerRegistryPath }}</mat-error>
+            </mat-form-field>
+          </ng-container>
+          <mat-form-field *ngIf="!showCustomDockerRegistryPath">
+            <input
+              matInput
+              id="dockerRegistryPathInput"
+              name="customDockerRegistryPath"
+              [(ngModel)]="customDockerRegistryPath"
+              maxlength="256"
+              matTooltip="Docker registry path"
+              required
+              readonly
+            />
+            <mat-hint>Read-only</mat-hint>
+          </mat-form-field>
+
+          <mat-checkbox 
+            class="mb-2"
+            [disabled]="disablePrivateCheckbox"
+            name="tool.private_access"
+            [(ngModel)]="tool.private_access"
+            (change)="togglePrivateAccess()"
+            matTooltip="Image registry entry is private."
+            labelPosition="before"
+            id="privateTool"
+          > Private Image:</mat-checkbox>
+          <mat-form-field *ngIf="tool.private_access">
+            <mat-label>Tool Maintainer Email</mat-label>
+            <input
+              matInput
+              id="toolMaintainerEmailInput"
+              type="email"
+              name="email"
+              [(ngModel)]="tool.tool_maintainer_email"
+              required
+              maxlength="256"
+              matTooltip="Email of the tool maintainer"
+              placeholder="e.g. example@domain.com"
+            />
+            <mat-error *ngIf="formErrors.email">{{ formErrors.email }}</mat-error>
+            <mat-hint>Email of the person responsible for giving users access to your tool on external sites</mat-hint>
+          </mat-form-field>
+          
+          <mat-form-field>
+            <mat-label>Tool Name</mat-label>
+            <input
+              matInput
+              type="text"
+              name="toolName"
+              [(ngModel)]="tool.toolname"
+              maxlength="256"
+              [pattern]="validationPatterns.toolName"
+              matTooltip="Optional Dockstore tool path suffix"
+              placeholder="e.g. liftover-fast (optional)"
+            />
+            <mat-error *ngIf="formErrors.toolName">{{ formErrors.toolName }}</mat-error>
+            <mat-hint>Name to distinguish between multiple tools within the same repository</mat-hint>
+          </mat-form-field>
```

Comment:
This comment for line 345/339 because I can't comment there directly -- was like this, but opened in WebStorm and `savingActive` doesn't exist.